### PR TITLE
(pkgsStatic.perl): Patch Perl to avoid duplicate symbols with musl 

### LINKED ISSUE #295608: Build failure: `pkgsStatic.perl` (Perl interpreter statically linked to Musl libc)
### Steps To Reproduce

1. e.g. `nix build -I nixpkgs=flake:github:NixOS/nixpkgs/23.11 nixpkgs#pkgsStatic.perl538`

It still affects `master` as of 2024-03-12. edit: Changed title for clarity; `pkgsMusl.perl` is fine.

### Build log

<details>
<summary>tail of build log</summary>

```
make -C dist/threads-shared/ PERL_CORE=1 LIBPERL=libperl.a LINKTYPE=static static
make[2]: Entering directory '/build/perl-5.38.2/dist/threads-shared'
"../../miniperl_top" "-I../../lib" "../../lib/ExtUtils/xsubpp"  -typemap '/build/perl-5.38.2/dist/threads-shared/../../lib/ExtUtils/typemap'  shared.xs > shared.xsc
mv shared.xsc shared.c
x86_64-unknown-linux-musl-gcc -c   -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -Wno-use-after-free -fwrapv -fno-strict-aliasing   -DVERSION=\"1.68\" -DXS_VERSION=\"1.68\" -fPIC -Wno-unused-function "-I../.."   shared.c
In file included from /nix/store/5m5zr8z4m8wx4v2c6w3fwcg07igrwzjy-libxcrypt-static-x86_64-unknown-linux-musl-4.4.36/include/crypt.h:22,
                 from ../../reentr.h:124,
                 from ../../op.h:700,
                 from ../../perl.h:4542,
                 from shared.xs:131:
/nix/store/0nfn0979vnb1w48rzm5mkx9zvbrkmxpp-x86_64-unknown-linux-musl-gcc-13.2.0/x86_64-unknown-linux-musl/sys-include/sys/cdefs.h:1:2: warning: #warning usage of non-standard #include <sys/cdefs.h> is deprecated [-Wcpp]
    1 | #warning usage of non-standard #include <sys/cdefs.h> is deprecated
      |  ^~~~~~~
In file included from ../../perl.h:7816:
../../sv_inline.h:566:1: warning: 'always_inline' function might not be inlinable [-Wattributes]
  566 | Perl_SvPVXtrue(pTHX_ SV *sv)
      | ^~~~~~~~~~~~~~
../../sv_inline.h:536:1: warning: 'always_inline' function might not be inlinable [-Wattributes]
  536 | Perl_newSV_type_mortal(pTHX_ const svtype type)
      | ^~~~~~~~~~~~~~~~~~~~~~
rm -f "../../lib/auto/threads/shared/shared.a"
x86_64-unknown-linux-musl-ar cr "../../lib/auto/threads/shared/shared.a" shared.o
x86_64-unknown-linux-musl-ranlib "../../lib/auto/threads/shared/shared.a"
chmod 755 ../../lib/auto/threads/shared/shared.a
make[2]: Leaving directory '/build/perl-5.38.2/dist/threads-shared'
./miniperl_top statars B Devel::Peek Fcntl File::DosGlob File::Glob Hash::Util Hash::Util::FieldHash I18N::Langinfo Opcode POSIX PerlIO::encoding PerlIO::mmap PerlIO::scalar PerlIO::via SDBM_File Sys::Hostname XS::APItest XS::Typemap attributes mro re Compress::Raw::Bzip2 Compress::Raw::Zlib Digest::MD5 Digest::SHA Encode Filter::Util::Call IPC::SysV MIME::Base64 Math::BigInt::FastCalc List::Util Socket Sys::Syslog Time::Piece Unicode::Collate Data::Dumper Devel::PPPort IO Cwd Storable Time::HiRes Unicode::Normalize threads threads::shared > static.list
./miniperl_top extlibs B Devel::Peek Fcntl File::DosGlob File::Glob Hash::Util Hash::Util::FieldHash I18N::Langinfo Opcode POSIX PerlIO::encoding PerlIO::mmap PerlIO::scalar PerlIO::via SDBM_File Sys::Hostname XS::APItest XS::Typemap attributes mro re Compress::Raw::Bzip2 Compress::Raw::Zlib Digest::MD5 Digest::SHA Encode Filter::Util::Call IPC::SysV MIME::Base64 Math::BigInt::FastCalc List::Util Socket Sys::Syslog Time::Piece Unicode::Collate Data::Dumper Devel::PPPort IO Cwd Storable Time::HiRes Unicode::Normalize threads threads::shared > ext.libs
x86_64-unknown-linux-musl-gcc -Wl,-E -o perl perlmain.o libperl.a lib/auto/B/B.a lib/auto/Devel/Peek/Peek.a lib/auto/Fcntl/Fcntl.a lib/auto/File/DosGlob/DosGlob.a lib/auto/File/Glob/Glob.a lib/auto/Hash/Util/Util.a lib/auto/Hash/Util/FieldHash/FieldHash.a lib/auto/I18N/Langinfo/Langinfo.a lib/auto/Opcode/Opcode.a lib/auto/POSIX/POSIX.a lib/auto/PerlIO/encoding/encoding.a lib/auto/PerlIO/mmap/mmap.a lib/auto/PerlIO/scalar/scalar.a lib/auto/PerlIO/via/via.a lib/auto/SDBM_File/SDBM_File.a lib/auto/Sys/Hostname/Hostname.a lib/auto/XS/APItest/APItest.a lib/auto/XS/Typemap/Typemap.a lib/auto/attributes/attributes.a lib/auto/mro/mro.a lib/auto/re/re.a lib/auto/Compress/Raw/Bzip2/Bzip2.a lib/auto/Compress/Raw/Zlib/Zlib.a lib/auto/Digest/MD5/MD5.a lib/auto/Digest/SHA/SHA.a lib/auto/Encode/Encode.a lib/auto/Filter/Util/Call/Call.a lib/auto/IPC/SysV/SysV.a lib/auto/MIME/Base64/Base64.a lib/auto/Math/BigInt/FastCalc/FastCalc.a lib/auto/List/Util/Util.a lib/auto/Socket/Socket.a lib/auto/Sys/Syslog/Syslog.a lib/auto/Time/Piece/Piece.a lib/auto/Unicode/Collate/Collate.a lib/auto/Data/Dumper/Dumper.a lib/auto/Devel/PPPort/PPPort.a lib/auto/IO/IO.a lib/auto/Cwd/Cwd.a lib/auto/Storable/Storable.a lib/auto/Time/HiRes/HiRes.a lib/auto/Unicode/Normalize/Normalize.a lib/auto/threads/threads.a lib/auto/threads/shared/shared.a -lm -lcrypt -lpthread -ldl -lm -L/nix/store/ljqvlm46wa0v3r39kii8k2gzwk6jxybs-zlib-static-x86_64-unknown-linux-musl-1.3.1/lib -lz -lrt -lpthread
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_reg_add_data':
re_comp.c:(.text+0xa390): multiple definition of `Perl_reg_add_data'; libperl.a(regcomp.o):regcomp.c:(.text+0x70a0): first defined here
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_populate_anyof_bitmap_from_invlist':
re_comp.c:(.text+0xa8e0): multiple definition of `Perl_populate_anyof_bitmap_from_invlist'; libperl.a(regcomp.o):regcomp.c:(.text+0x7b60): first defined here
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_add_above_Latin1_folds':
re_comp.c:(.text+0xab10): multiple definition of `Perl_add_above_Latin1_folds'; libperl.a(regcomp.o):regcomp.c:(.text+0x7ce0): first defined here
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_set_ANYOF_arg':
re_comp.c:(.text+0xb020): multiple definition of `Perl_set_ANYOF_arg'; libperl.a(regcomp.o):regcomp.c:(.text+0x8160): first defined here
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_get_ANYOFM_contents':
re_comp.c:(.text+0xcf50): multiple definition of `Perl_get_ANYOFM_contents'; libperl.a(regcomp.o):regcomp.c:(.text+0x9d70): first defined here
/nix/store/sqii1hjj7rr1d5c5wwxlswqrv39nispr-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_get_ANYOFHbbm_contents':
re_comp.c:(.text+0xd020): multiple definition of `Perl_get_ANYOFHbbm_contents'; libperl.a(regcomp.o):regcomp.c:(.text+0x9e40): first defined here
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:154: perl] Error 1
make[1]: Leaving directory '/build/perl-5.38.2'
make: *** [Makefile:87: all] Error 2
```
</details>

[full log](https://gist.github.com/YoshiRulz/9386c76c8d2a59deb74a0edd114f1327)

### Additional context

I'm aiming for `pkgsStatic.wine64Packages.minimal`, which may be a fool's errand but it's worth a try.

### Notify maintainers

@stigtsp @zakame @dasJ (and Eelco?)

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.38, NixOS, 23.05 (Stoat), 23.05.1875.8df7a67abaf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by bltavares**
## Description of changes

This commit backports a commit introduced on Perl 5.39.2 to allow `pkgs.pkgsStatic.perl` to compile without raising errors about duplicate symbols being defined during the linking process.

This fixes https://github.com/NixOS/nixpkgs/issues/295608

| Tested on | Status |
| --- | --- |
| M1 Mac (aarch64-darwin) | Failed due to other issues* |
| NixOS VM (x86_64-linux) | ✅  |
| Nix (aarch64-linux) | ✅  |

**Test method**:

```sh
nix repl -I nixpkgs=$PWD
nix repl> pkgs = import <nixpkgs> {}
nix repl> :b pkgs.perl
nix repl> :b pkgs.pkgsStatic.perl
```

Despite the upstream commit being scoped to all environments, we are only applying this patch only on the `pkgsStatic` context to avoid rebuilding all of NixOS packages by changing Perl.

It includes an additional build flag [also used upstream](https://github.com/Perl/perl5/pull/22052)
for configuring static builds.

*MacOS: `pkgsStatic.perl` still fails to compile after this change due to MacOS case-insensitiviness affecting the build recipe under /tmp, as `perl-cross/configure` is copied over `perl/Configure` file, as well as not finding `readelf` for the `configurePhase`.

https://github.com/NixOS/nixpkgs/issues/113985#issuecomment-785024419 https://github.com/NixOS/nix/issues/1446
https://github.com/NixOS/nix/issues/9318
https://github.com/NixOS/nix/issues/8995

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bltavares** [comment]
Despite the patch applying cleanly, the error still persists:

<details>
<summary>Error log</summary>

```
error: builder for '/nix/store/yc75djfdw129399igyshacynhk7zsfnf-perl-static-x86_64-unknown-linux-musl-5.38.2.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/826v1ayjaayg7j61c8mndfqmh110dxa6-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_set_ANYOF_arg':
       > re_comp.c:(.text+0xb020): multiple definition of `Perl_set_ANYOF_arg'; libperl.a(regcomp.o):regcomp.c:(.text+0x8160): first defined here
       > /nix/store/826v1ayjaayg7j61c8mndfqmh110dxa6-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_get_ANYOFM_contents':
       > re_comp.c:(.text+0xcf50): multiple definition of `Perl_get_ANYOFM_contents'; libperl.a(regcomp.o):regcomp.c:(.text+0x9d70): first defined here
       > /nix/store/826v1ayjaayg7j61c8mndfqmh110dxa6-x86_64-unknown-linux-musl-binutils-2.41/bin/x86_64-unknown-linux-musl-ld: lib/auto/re/re.a(re_comp.o): in function `Perl_get_ANYOFHbbm_contents':
       > re_comp.c:(.text+0xd020): multiple definition of `Perl_get_ANYOFHbbm_contents'; libperl.a(regcomp.o):regcomp.c:(.text+0x9e40): first defined here
       > collect2: error: ld returned 1 exit status
       > make[1]: *** [Makefile:154: perl] Error 1
       > make[1]: Leaving directory '/build/perl-5.38.2'
       > make: *** [Makefile:87: all] Error 2
       For full logs, run 'nix log /nix/store/yc75djfdw129399igyshacynhk7zsfnf-perl-static-x86_64-unknown-linux-musl-5.38.2.drv'.
```
</details>

I'll need to further investigate upstream what else can be done

File: pkgs/development/interpreters/perl/intepreter.nix [hunk 1]
```diff
@@ -110,7 +113,7 @@ stdenv.mkDerivation (rec {
     ++ lib.optional stdenv.isSunOS "-Dcc=gcc"
     ++ lib.optional enableThreading "-Dusethreads"
     ++ lib.optional (!enableCrypt) "-A clear:d_crypt_r"
-    ++ lib.optional stdenv.hostPlatform.isStatic "--all-static"
+    ++ lib.optionals stdenv.hostPlatform.isStatic [ "--all-static" "-Uusedl" ]
```
Comment (bltavares) [inline pkgs/development/interpreters/perl/intepreter.nix:116]: I'm not very familiar with this flag, but it seems to be used upstream and produces a binary in the end.

**Comment by YoshiRulz** [comment]
Can confirm the interpreter is linked statically with this PR. Thank you!

**Review by stigtsp — CHANGES_REQUESTED** [review]
Please make sure the commit message (and PR title) matches [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)

For example, we usually use the attribute name as prefix (without parens), like: `perl: ...`

**Comment by stigtsp** [comment]
I've checked that https://github.com/Perl/perl5/commit/ba6e2c38aafc23cf114f3ba0d0ff3baead34328b (merged into blead https://github.com/Perl/perl5/pull/21320) is the same as static-stdenv-5.39.2.patch vendored into this PR.

**Comment by stigtsp** [comment]
I'm getting the segfault error when running `perl -V`

```
$ set c (nix-build -A pkgsStatic.perl); and $c/bin/perl -V
fish: Job 1, 'result/bin/perl -V' terminated by signal SIGSEGV (Address boundary error)
```

dmesg output:
```
[285797.506144] perl[403625]: segfault at 5878ed7 ip 00000000004a5072 sp 00007ffd945de7c0 error 4 in perl[401000+2d3000] likely on CPU 11 (core 6, socket 0)
[285797.506158] Code: 0f b6 4a 01 49 8d 04 82 4c 39 d0 48 0f 45 f0 48 89 74 24 28 48 8b 44 24 10 c7 44 24 08 f6 ff ff ff 4c 8b 48 10 4d 39 cb 73 08 <41> 0f b6 03 89 44 24 08 81 f9 99 00 00 00 0f 87 fa 6c 00 00 48 8d
```

Would like to check this one out a bit before merging

**Comment by YoshiRulz** [comment]
CNR at 63d7ba59d7d6. It consistently builds to `/nix/store/41nylr1jx656aki1xhw5x44fp4c36l8s-perl-static-x86_64-unknown-linux-musl-5.38.2` on my machine.

**Comment by YoshiRulz** [comment]
That command does segfault on my machine, but if I launch a shell `nix shell -ivL github:NixOS/nixpkgs/63d7ba59d7d67d07b60729c2fb3b2f6653ebc345#pkgsStatic.perl` then it's fine.

**Comment by jcaesar** [comment]
To dig into this a little bit, I got myself a debuggable version of perl with
```bash
nix shell -vL --expr 'let pkgs = import (builtins.getFlake github:NixOS/nixpkgs/63d7ba59d7d67d07b60729c2fb3b2f6653ebc345) { system = "x86_64-linux"; }; in pkgs.pkgsStatic.perl.overrideAttrs { env.NIX_CFLAGS_COMPILE = "-g -O0"; dontStrip = true; }'
```

Of course, the segv doesn't happen in valgrind or gdb, but I can get a stacktrace with `machinectl gdb`:
```
#0  0x00000000005780c3 in S_regmatch (my_perl=0x7f6a271c6030, reginfo=0x7ffcd1700cb0,
    startpos=0x7f6a270a9f58 "\npackage='perl5'\npager='less'\npasscat='cat /etc/passwd'\npatchlevel='38'\npath_sep=':'\nperl=''\nperl5=''\nperl_lc_all_category_positions_init=''\nperl_lc_all_separator=''\nperl_patchlevel=''\nperl_static_inl"..., prog=0x7f6a27158684) at regexec.c:6560
#1  0x000000000056df16 in S_regtry (my_perl=0x7f6a271c6030, reginfo=0x7ffcd1700cb0, startposp=0x7ffcd1700bc8) at regexec.c:4407
#2  0x000000000056d6b7 in Perl_regexec_flags (my_perl=0x7f6a271c6030, rx=0x7f6a27145358,
    stringarg=0x7f6a270a2180 "\nAuthor=''\nCONFIG='true'\nDate='$Date'\nHeader=''\nId='$Id'\nLocker=''\nLog='$Log'\nPATCHLEVEL='38'\nPERL_API_REVISION='5'\nPERL_API_SUBVERSION='0'\nPERL_API_VERSION='38'\nPERL_CONFIG_SH='true'\nPERL_PATCHLEVEL="..., strend=0x7f6a270ac236 "",
    strbeg=0x7f6a270a2180 "\nAuthor=''\nCONFIG='true'\nDate='$Date'\nHeader=''\nId='$Id'\nLocker=''\nLog='$Log'\nPATCHLEVEL='38'\nPERL_API_REVISION='5'\nPERL_API_SUBVERSION='0'\nPERL_API_VERSION='38'\nPERL_CONFIG_SH='true'\nPERL_PATCHLEVEL="..., minend=0, sv=0x7f6a271802a0, data=0x0,
    flags=97) at regexec.c:4117
#3  0x0000000000510c62 in Perl_pp_match (my_perl=0x7f6a271c6030) at pp_hot.c:3172
#4  0x000000000063411a in Perl_runops_standard (my_perl=0x7f6a271c6030) at run.c:41
#5  0x000000000044a954 in Perl_call_sv (my_perl=0x7f6a271c6030, sv=0x7f6a271452c8, flags=4098) at perl.c:3133
#6  0x00000000004af632 in Perl_magic_methcall (my_perl=0x7f6a271c6030, sv=0x7f6a27145388, mg=0x7f6a270dd5e0, meth=0x7f6a271452c8, flags=0, argc=0)
    at mg.c:2141
#7  0x00000000004af916 in S_magic_methcall1 (my_perl=0x7f6a271c6030, sv=0x7f6a27145388, mg=0x7f6a270dd5e0, meth=0x7f6a271452c8, flags=0, n=1,
    val=0x0) at mg.c:2175
#8  0x00000000004af95e in S_magic_methpack (my_perl=0x7f6a271c6030, sv=0x7f6a27145388, mg=0x7f6a270dd5e0, meth=0x7f6a271452c8) at mg.c:2185
#9  0x00000000004afa23 in Perl_magic_getpack (my_perl=0x7f6a271c6030, sv=0x7f6a27145388, mg=0x7f6a270dd5e0) at mg.c:2198
#10 0x00000000004aa6b5 in Perl_mg_get (my_perl=0x7f6a271c6030, sv=0x7f6a27145388) at mg.c:198
#11 0x0000000000514bf9 in Perl_pp_multideref (my_perl=0x7f6a271c6030) at pp_hot.c:4032
#12 0x000000000063411a in Perl_runops_standard (my_perl=0x7f6a271c6030) at run.c:41
#13 0x000000000044a098 in S_run_body (my_perl=0x7f6a271c6030, oldscope=1) at perl.c:2812
#14 0x0000000000449a94 in perl_run (my_perl=0x7f6a271c6030) at perl.c:2727
#15 0x000000000040865e in main (argc=2, argv=0x7ffcd17016c8, env=0x7ffcd17016e0) at perlmain.c:127
```

I was trying to find out what code causes the segv but didn't get very far: `-V:ld` seems to run something like
```perl
use Config; Config::config_vars(ld)
```
but trying to trace to the code with `-d:Trace` (requires `.withPackages (ps: [ps.DevelTrace])`), it segvs in some completely different place, so I kind of doubt that "It seems to function as an interpreter though".

(At this point, I lost interest in this PR, because I found a different way to build a static qemu.)

**Comment by YoshiRulz** [comment]
Superceded by #338838.

**Pull request closed without merging**